optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-backend/src/main/java/org/optaplanner/workbench/screens/domaineditor/backend/server/PlannerDomainHandler.java

```diff
@@ -66,15 +66,17 @@ public void setDefaultValues( DataObject dataObject, Map<String, Object> options
 
     @Override
     public void processDataObject( DataObject dataObject, DataModel dataModel ) {
-        boolean hasPlanningEntity = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION ) != null;
-        boolean hasComparatorDefinition = dataObject.getAnnotation( ComparatorDefinition.class.getName() ) != null;
-        if ( hasPlanningEntity && hasComparatorDefinition ) {
-            try {
-                ComparatorObject comparatorObject = plannerDataObjectEditorService.extractComparatorObject( dataObject, dataModel );
-                dataObject.getNestedClasses().clear();
-                dataObject.getNestedClasses().add( comparatorObject );
-            } catch ( Exception e ) {
-                logger.error( "Unable to extract comparator object from " + dataObject.getClassName() );
+        if ( dataModel != null ) {
+            boolean hasPlanningEntity = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION ) != null;
+            boolean hasComparatorDefinition = dataObject.getAnnotation( ComparatorDefinition.class.getName() ) != null;
+            if ( hasPlanningEntity && hasComparatorDefinition ) {
+                try {
+                    ComparatorObject comparatorObject = plannerDataObjectEditorService.extractComparatorObject( dataObject, dataModel );
+                    dataObject.getNestedClasses().clear();
+                    dataObject.getNestedClasses().add( comparatorObject );
+                } catch ( Exception e ) {
```

Comment:
@ge0ffrey `plannerDataObjectEditorService.extractComparatorObject` throws `IllegalStateException` under certain circumstances (e.g. unexpected data object state, etc.). I think it makes sense to introduce checked service exception here. I'll update the PR.